NEWS.md

```diff
@@ -28,6 +28,8 @@
 * New `backport_linter()` for detecting mismatched R version dependencies (#506, @MichaelChirico)
 * `paren_brace_linter` and `no_tab_linter` also use more reliable matching (e.g.,
   excluding matches found in comments; #441 and #545, @russHyde)
+* Lints are now marked with the name of the `linter` that caused them instead of the name of their implementation
+  function (#664, #673, @AshesITR).
```

Comment:
I think adding some more context (maybe an extra sentence) in the NEWS will be helpful -- I guess most existing users would learn about this by checking the NEWS on release, so a quick motivating example here would be good.

Reply:
WDYT about making a bigger news all around the combination of this with #660?

The more I think about it the more I think it's worth a vignette...

Reply:
Agreed. I'll write a note there.